templates/orders/downloadable.php

```diff
@@ -5,72 +5,57 @@
 
     <div class="panel-group" id="accordion">
         <?php
-        $download_permissions = $wpdb->get_results(
-            $wpdb->prepare(
-                "
-                SELECT * FROM {$wpdb->prefix}woocommerce_downloadable_product_permissions
-                WHERE order_id = %d ORDER BY product_id
-            ", $order->get_id()
-            )
-        );
+        $data_store           = WC_Data_Store::load( 'customer-download' );
+        $download_permissions = array();
+        if ( 0 !== $order->get_id() ) {
+            $download_permissions = $data_store->get_downloads(
+                array(
+                    'order_id' => $order->get_id(),
+                    'orderby'  => 'product_id',
+                )
+            );
+        }
 
-        $product    = null;
-        $loop       = 0;
-        $file_count = 1;
+        $product      = null;
+        $loop         = 0;
+        $file_counter = 1;
 
         if ( $download_permissions && count( $download_permissions ) > 0 ) {
             foreach ( $download_permissions as $download ) {
-                if ( ! $product || $product->get_id() !== (int) $download->product_id ) {
-                    $product    = wc_get_product( absint( $download->product_id ) );
-                    $file_count = 1;
+                if ( ! $product || $product->get_id() !== $download->get_product_id() ) {
+                    $product      = wc_get_product( $download->get_product_id() );
+                    $file_counter = 1;
                 }
 
-                // don't show permissions to files that have since been removed
-                if ( ! $product || ! $product->exists() || ! $product->has_file( $download->download_id ) ) {
+                // don't show permissions to files that have since been removed.
+                if ( ! $product || ! $product->exists() || ! $product->has_file( $download->get_download_id() ) ) {
                     continue;
                 }
 
+                // Show file title instead of count if set.
+                $file = $product->get_file( $download->get_download_id() );
+                // translators: file name.
+                $file_count = isset( $file['name'] ) ? $file['name'] : sprintf( __( 'File %d', 'dokan-lite' ), $file_counter );
+
                 include 'order-download-permission-html.php';
 
-                $loop++;
-                $file_count++;
+                ++$loop;
+                ++$file_counter;
```

Comment:
**Findings:**
- `$file_count` is expected in `order-download-permission-html.php` as a numeric value.
- ` sprintf( __( 'File %d', 'dokan-lite' ), $file_counter )` is misleading the message in `order-download-permission-html.php` in **Line:14-26**